### PR TITLE
Updating Titles and Headings

### PR DESCRIPTION
**Pull request by carrielaben-da**
Conforming titles and headings to best practices:

- Making them descriptive and non-repetitive
- Title case for first two levels of headings on a given page, sentence case for lower levels (in keeping with DA style guide)
- Typo fixes

[CHANGELOG_BEGIN]

[CHANGELOG_END]

### CODE REVIEW

File: docs/source/app-dev/grpc/error-codes.rst [hunk 1]
```diff
@@ -8,11 +8,11 @@
 :tocdepth: 2
 
 
-Error Codes
-###########
+Understanding Error Codes
+#########################
```
Comment (stefanobaghino-da) [inline docs/source/app-dev/grpc/error-codes.rst:12]:
```suggestion
Error Codes
###########
```

In a sense, all documentation pages are about understanding something. Are you sure we need this in this title? The title seems pretty simple and self-explanatory as it were.

File: docs/source/app-dev/grpc/index.rst [hunk 2]
```diff
@@ -3,25 +3,25 @@
 
 .. _grpc:
 
-gRPC
-####
+Using gRPC with the Ledger API
+##############################
```
Comment (stefanobaghino-da) [inline docs/source/app-dev/grpc/index.rst:7]:
```suggestion
Using the Ledger API with gRPC
##############################
```

gRPC is the framework used to implement the Ledger API. What developers do is accessing the Ledger API using the tools provided by the gRPC framework (mostly code generation on the client-side). This wording seems more to the point, but feel free to object.

File: docs/source/concepts/ledger-model/index.rst [hunk 3]
```diff
@@ -3,8 +3,8 @@
 
 .. _da-ledgers:
 
-Daml Ledger Model
-#################
+Understanding the Daml Ledger Model
+###################################
```
Comment (stefanobaghino-da) [inline docs/source/concepts/ledger-model/index.rst:7]:
```suggestion
Daml Ledger Model
#################
```
I don't think it's necessary to add "Understanding the" here. It feels more natural and simple. What is this page about? The "Daml Ledger Model", that's what it says in the title.

File: docs/source/daml/intro/10_Functional101.rst [hunk 4]
```diff
@@ -1,7 +1,7 @@
 .. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-10 Functional Programming 101
+Functional Programming 101
 =============================
```
Comment (stefanobaghino-da) [inline docs/source/daml/intro/10_Functional101.rst:5]:
```suggestion
Functional Programming 101
==========================
```

File: docs/source/daml/intro/12_Testing.rst [hunk 5]
```diff
@@ -1,7 +1,7 @@
 .. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-12 Testing Daml Contracts
+Testing Daml Contracts
 =========================
```
Comment (stefanobaghino-da) [inline docs/source/daml/intro/12_Testing.rst:5]:
```suggestion
Testing Daml Contracts
======================
```

File: docs/source/daml/intro/3_Data.rst [hunk 6]
```diff
@@ -1,7 +1,7 @@
 .. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-3 Data types
+Data Types
 ============
```
Comment (stefanobaghino-da) [inline docs/source/daml/intro/3_Data.rst:5]:
```suggestion
Data Types
=========
```

File: docs/source/daml/intro/4_Transformations.rst [hunk 7]
```diff
@@ -1,7 +1,7 @@
 .. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-4 Transforming data using choices
+Transforming Data using Choices
 =================================
```
Comment (stefanobaghino-da) [inline docs/source/daml/intro/4_Transformations.rst:5]:
```suggestion
Transforming Data using Choices
===============================
```

File: docs/source/daml/intro/5_Restrictions.rst [hunk 8]
```diff
@@ -1,7 +1,7 @@
 .. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-5 Adding constraints to a contract
+Adding Constraints to a Contract
 ==================================
```
Comment (stefanobaghino-da) [inline docs/source/daml/intro/5_Restrictions.rst:5]:
```suggestion
Adding Constraints to a Contract
================================
```

File: docs/source/daml/intro/7_Composing.rst [hunk 9]
```diff
@@ -1,7 +1,7 @@
 .. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-7 Composing choices
+Composing Choices
 ===================
```
Comment (stefanobaghino-da) [inline docs/source/daml/intro/7_Composing.rst:5]:
```suggestion
Composing Choices
=================
```

File: docs/source/daml/intro/7_Composing.rst [hunk 10]
```diff
@@ -100,7 +100,7 @@ The project both changes and adds to the ``Iou`` model presented in :doc:`6_Part
   With the ``Iou`` model, an ``issuer`` could end up owing cash to anyone as transfers were authorized by just ``owner`` and ``newOwner``. In this project, only parties having an ``AssetHolder`` contract can end up owning assets. This allows the ``issuer`` to determine which parties may own their assets.
 - The ``Trade`` template adds a swap of two assets to the model.
 
-Composed choices and scripts
+Composed Choices and Scripts
 ------------------------------
```
Comment (stefanobaghino-da) [inline docs/source/daml/intro/7_Composing.rst:104]:
```suggestion
Composed Choices and Scripts
----------------------------
```

File: docs/source/daml/intro/8_Exceptions.rst [hunk 11]
```diff
@@ -1,7 +1,7 @@
 .. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-8 Exception Handling
+Exception Handling
 ====================
```
Comment (stefanobaghino-da) [inline docs/source/daml/intro/8_Exceptions.rst:5]:
```suggestion
Exception Handling
==================
```

File: docs/source/daml/intro/9_Dependencies.rst [hunk 12]
```diff
@@ -1,7 +1,7 @@
 .. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-9 Working with Dependencies
+Working with Dependencies
 ===========================
```
Comment (stefanobaghino-da) [inline docs/source/daml/intro/9_Dependencies.rst:5]:
```suggestion
Working with Dependencies
=========================
```

File: docs/source/daml/patterns/legends.rst [hunk 13]
```diff
@@ -1,7 +1,7 @@
 .. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-Diagram legends
+Diagram Legends
 ################
```
Comment (stefanobaghino-da) [inline docs/source/daml/patterns/legends.rst:5]:
```suggestion
Diagram Legends
###############
```

File: docs/source/daml/reference/index.rst [hunk 14]
```diff
@@ -1,8 +1,8 @@
 .. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-Language reference docs
-#######################
+Language Reference Documents
+############################
```
Comment (stefanobaghino-da) [inline docs/source/daml/reference/index.rst:5]:
```suggestion
Language Reference
##################
```

Probably simpler.

File: docs/source/getting-started/installation.rst [hunk 15]
```diff
@@ -4,7 +4,7 @@
 Installing the SDK
 ##################
 
-1. Installing the Dependencies
+Installing the Dependencies
 ******************************
```
Comment (stefanobaghino-da) [inline docs/source/getting-started/installation.rst:8]:
```suggestion
Installing the Dependencies
***************************
```

File: docs/source/getting-started/installation.rst [hunk 16]
```diff
@@ -16,7 +16,7 @@ You need to install:
 
    As part of the installation process you may need to set up the ``JAVA_HOME`` variable. You can find instructions for this process on :doc:`Windows,macOS, and Linux here <path-variables>`.
 
-2. Choosing Daml Enterprise or Daml Open Source
+Choosing Daml Enterprise or Daml Open Source
 ***********************************************
```
Comment (stefanobaghino-da) [inline docs/source/getting-started/installation.rst:20]:
```suggestion
Choosing Daml Enterprise or Daml Open Source
********************************************
```

File: docs/source/json-api/index.rst [hunk 17]
```diff
@@ -205,7 +205,7 @@ start the standalone JAR, you can use the following command:
 Replace the version number ``2.0.0`` by the version of the SDK you are
 using.
 
-With Query Store
+With query store
 ------------------
```
Comment (stefanobaghino-da) [inline docs/source/json-api/index.rst:209]:
```suggestion
With query store
----------------
```

File: docs/source/tools/trigger-service/authorization.rst [hunk 18]
```diff
@@ -76,7 +76,7 @@ Alternatively, it can append a custom ``redirect_url`` parameter to the login UR
 Note that login with the IAM may require entering credentials into a web-form,
 i.e. the login URL should be opened in a web browser.
 
-Example Usage
+Example
 *************
```
Comment (stefanobaghino-da) [inline docs/source/tools/trigger-service/authorization.rst:80]:
```suggestion
Example
*******
```

File: docs/source/upgrade/upgrade.rst [hunk 19]
```diff
@@ -170,7 +170,7 @@ Now we can build and deploy ``carbon-1.0.0``.
    $ daml ledger upload-dar --port 6865
 
 
-Create some carbon-1.0.0 certificates
+Creating carbon-1.0.0 Certificates
 =====================================
```
Comment (stefanobaghino-da) [inline docs/source/upgrade/upgrade.rst:174]:
```suggestion
Creating carbon-1.0.0 Certificates
==================================
```

**Review by stefanobaghino-da — APPROVED** [review]
Mostly minor things.

One important thing though: I find the "title case for top-2 level headers, sentence case for everything else" a bit confusing. I realized after I was done with the review that a lot of my comments were not accounting for that. Even then, it's quite difficult to tell whether the right case was used in the right position, unless you memorized what heading we are  using at what level (which RST doesn't help with, since there's not check that a specific hierarchy is used). I believe it would be easier to have developers contribute content that requires less proof-reading if we stick to a single style. Personally I'm a fan of sentence case: it's the same you use in a sentence, no special rule to remember, just write in the same way you always did. I'm not too attached to it either, but I would still favor sticking to a single title capitalization style to make things simpler to review. The "top 2" categorization is also quite arbitrary and not all our documentation pages require the same level of nesting.

We don't have to decide now or make changes to this PR, but I'd like you to think about this. Thanks!

**Comment by stefanobaghino-da** [comment]
@carrielaben-da How is this going on?

File: docs/source/daml/intro/10_Functional101.rst [hunk 20]
```diff
@@ -289,7 +289,7 @@ If we need functions that can return two (or more) types of things we need to en
   :start-after: -- INT_OR_TEXT_BEGIN
   :end-before: -- INT_OR_TEXT_END
 
-Branching in actions
+Branching In Actions
```
Comment (stefanobaghino-da) [inline docs/source/daml/intro/10_Functional101.rst:292]: Should "in" be capitalized or not? I saw both.

File: docs/source/daml/troubleshooting.rst [hunk 21]
```diff
@@ -185,7 +185,7 @@ Now that the contract is bound to a variable, we can check whether it has some e
 
 .. _faqs-must-fail:
 
-How to test that an update action cannot be committed
+How To Test That an Update Action Cannot Be Committed
```
Comment (stefanobaghino-da) [inline docs/source/daml/troubleshooting.rst:188]: Is "to" supposed to be capitalized or not? I see both.

**Review by stefanobaghino-da — APPROVED** [review]
All excellent!

File: docs/source/daml/troubleshooting.rst (hunk 21 above)
Comment (carrielaben-da) [reply docs/source/daml/troubleshooting.rst:188]: I checked this on a title case converter tool using AP style as the default, the results include To being capitalized (but not "In" above, I will change that).

File: docs/source/daml/intro/10_Functional101.rst [hunk 22]
```diff
@@ -134,7 +134,7 @@ Let's turn this into prose: Given that ``t`` is the type of a template, and that
 
 That's quite a mouthful, and does require one to know what *meaning* the typeclass ``Choice`` gives to parameters ``t`` ``c`` and ``r``, but in many cases, that's obvious from the context or names of typeclasses and variables.
 
-Pattern Matching in Arguments
+Pattern Matching In Arguments
```
Comment (carrielaben-da) [inline docs/source/daml/intro/10_Functional101.rst:137]:
```suggestion
Pattern Matching in Arguments
```

File: docs/source/daml/intro/10_Functional101.rst [hunk 23]
```diff
@@ -289,7 +289,7 @@ If we need functions that can return two (or more) types of things we need to en
   :start-after: -- INT_OR_TEXT_BEGIN
   :end-before: -- INT_OR_TEXT_END
 
-Branching in Actions
+Branching In Actions
```
Comment (carrielaben-da) [inline docs/source/daml/intro/10_Functional101.rst:292]:
```suggestion
Branching in Actions
```

File: docs/source/daml/intro/10_Functional101.rst [hunk 24]
```diff
@@ -459,7 +459,7 @@ If there is no explicit iterator, you can use recursion. Let's try to write a fu
 
 You may be tempted to make ``reverseWorker`` a local definition inside ``reverse``, but Daml only supports recursion for top-level functions so the recursive part ``recurseWorker`` has to be its own top-level function.
 
-Folds and Maps in Action Contexts
+Folds and Maps In Action Contexts
```
Comment (carrielaben-da) [inline docs/source/daml/intro/10_Functional101.rst:462]:
```suggestion
Folds and Maps in Action Contexts
```

File: docs/source/daml/reference/contract-keys.rst [hunk 25]
```diff
@@ -17,15 +17,15 @@ Here's an example of setting up a contract key for a bank account, to act as a b
    :start-after: -- start contract key setup snippet
    :end-before: -- end contract key setup snippet
 
-What can be a contract key
+What Can Be a Contract Key
 **************************
 
 The key can be an arbitrary serializable expression that does **not** contain contract IDs. However, it **must** include every party that you want to use as a ``maintainer`` (see `Specifying maintainers`_ below).
```
Comment (carrielaben-da) [inline docs/source/daml/reference/contract-keys.rst:23]:
```suggestion
The key can be an arbitrary serializable expression that does **not** contain contract IDs. However, it **must** include every party that you want to use as a ``maintainer`` (see `Specify Maintainers`_ below).
```

**Pull request merged**